From 43b2543a297aff1a15773fd77f46c7f2c7fc1d88 Mon Sep 17 00:00:00 2001
From: gregor herrmann <gregoa@debian.org>
Date: Thu, 8 Dec 2011 23:54:22 +0100
Subject: [PATCH 2/2] dh_perl: add perl:Current and perl::Next substvars in v9

Adds 2 new substitution variables, that expand to e.g. "perl (>= 5.14.2)"
and "perl (<< 5.14.3~)" for packages that rely on the exact perl version.

Closes: #579112
---
 dh_perl |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/dh_perl b/dh_perl
index bf2506c..9f1eec7 100755
--- a/dh_perl
+++ b/dh_perl
@@ -25,6 +25,10 @@ and will use this information to generate a dependency on B<perl> or
 B<perlapi>. The dependency will be substituted into your package's F<control>
 file wherever you place the token B<${perl:Depends}>.
 
+Starting with compatibility level 9, B<${perl:Current}> and B<${perl:Next}>
+are expanded to B<perl[-base] (E<gt>= $currentversion)> and B<perl[-base] (<<
+$nextminorversion~)>.
+
 B<dh_perl> also cleans up empty directories that MakeMaker can generate when
 installing Perl modules.
 
@@ -116,11 +120,16 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 
 	if ($deps) {
 		my $version="";
+		my $currentversion="";
+		my $nextversion="";
 		if ($deps & XS_MODULE or $dh{V_FLAG_SET}) {
 			($version) = `dpkg -s $perl` =~ /^Version:\s*(\S+)/m
 				unless $version;
 			$version = ">= $version";
 		}
+		$currentversion = ">= $Config{version}";
+		my ($v1, $v2, $v3) = split /\./, $Config{version};
+		$nextversion = "<< " . (join ".", $v1, $v2, $v3 + 1) . "~";
 		
 		# no need to depend on an un-versioned perl-base -- it's
 		# essential
@@ -131,6 +140,12 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 		addsubstvar($package, "perl:Depends",
 			"perlapi-" . ($Config{debian_abi} || $Config{version}))
 			if $deps & XS_MODULE;
+
+		# add perl:Current and perl:Next denoting the current and next possible minor version
+		addsubstvar($package, "perl:Current", $perl, $currentversion)
+			unless compat(8);
+		addsubstvar($package, "perl:Next", $perl, $nextversion)
+			unless compat(8);
 	}
 
 	# MakeMaker always makes lib and share dirs, but typically
-- 
1.7.7.3

